backend/core/helpers.py

```
@@ -888,20 +888,40 @@ def csf_functions(user):
-        (object_ids, _, _) = RoleAssignment.get_accessible_object_ids(
-            Folder.get_root_folder(), user, model
+    root_folder = Folder.get_root_folder()
+
+    # Fetch accessible object IDs for models
+    accessible_objects = {
+        model: set(
+            RoleAssignment.get_accessible_object_ids(root_folder, user, model)[0]
         )
-        return model.objects.filter(id__in=object_ids)
+        for model in [
+            AppliedControl,
```

Ruff (0.8.2)

`AppliedControl` may be undefined, or defined from star imports

(F405)